posts/2019/11/Files-Improvements.md

```diff
@@ -0,0 +1,77 @@
+
+![Single click](/images/spyder-files-single-click.png)
+
+Bear in mind that changing this configuration option will also affect the behaviour of the Project Explorer pane.
+
+## Open files externally
+
+Additionally, we added a context menu action called Open externally to open files with the Operating System default program associated with the file type. To use this action, go to the file you want to open in the files tab and right-click it to see the context menu. Then select “Open Externally” and the file will be opened outside Spyder, with the default program associated with its file type, depending on the Operating System.
+
+![Open Externally](/images/spyder-files-open-externally.png)
+
+Moreover, now it is possible to select several files to perform a specific action thanks to the multi-select functionality added. For this, press CTRL or SHIFT and Click on the files that are going to be selected without releasing the CTRL/SHIFT. It is also possible to select the files without doing it one by one, but selecting a file with the CTRL/SHIFT clicked and selecting a not contiguous file to this one which will select all the files between them.
+By doing this, it is possible to execute some of the actions available in the context menu for all the files selected including delete, rename, move, copy and open externally.
+
+
+![Select Files](/images/spyder-files-select-files.png)
+
+## Absolute and relative path handling
+
+Another functionality added was the ability to copy and paste files and their absolute or relative paths. To access these actions, go to the context menu by right-clicking a file in the files tab. 
+
+![Copy Path](/images/spyder-files-copy-path.png)
+
+With this, a file can be copied from the files tab and pasted anywhere else directly from Spyder. The “Copy Absolute Path” and “Copy Relative Path” actions, give us access to the files’ path, to be pasted as text. 
+
+This way, when pasting the absolute path of a file, we get the complete path of the file starting from the root.
+
+![Absolute Path](/images/spyder-files-absolute-path.png)
+
+When using the option “Copy Relative Path”, we get the path of the file relative to the directory in which we are located.
+
+![Relative Path](/images/spyder-files-relative-path.png)
+
+Finally, files are now displayed along with icons depending on their file type. There are different icons for each programming language file extension, including c, cpp, csharp, java, python, r, swift, for each different file type including jpg, mp3, m4a and other file extensions including txt and tex. This allows the users to visually identify what type of file they are seeing in the files pane, in order to select the correct associated applications to open these files.
```

Comment:
```suggestion
Finally, files in the `Files` pane are now displayed with icons depending on their type. There are custom icons for the source files of a variety of common programming languages including C C++, C Sharp, Java, Python, R, and Swift; for media types like JPEG, MP3, and M4A; and for other common extensions such as `.txt` and `.tex`. This allows you to identify file types with just a glance and select the appropriate application accordingly.
```

* Fix confusing mix of language names and extensions for programming language source files
* Organize list and make it make much more sense
* Fix bad commas and comma splices
* Clarify and simplify explanations and remove redundancy and awkwardness
* Fix grammar
* Use "you" phrasing to put the focus on the reader as the user